ddl/index.go

```diff
@@ -382,9 +384,33 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 		}
 		return ver, err
 	}
+	for _, hiddenCol := range hiddenCols {
+		columnInfo := model.FindColumnInfo(tblInfo.Columns, hiddenCol.Name.L)
+		if columnInfo != nil && columnInfo.State == model.StatePublic {
+			// We already have a column with the same column name.
```

Comment:
1. Em, try to not use We, I, You, etc.
2. I think the `same name check`( include index same check above) should be brought forward, in ddl worker is a little bit late.

Reply:
1. Copy from func checkAddColumn, and we have many usages with `We`. 
2. For index name, see L3588 in ddl.api; for hidden column, see L3508

Reply:
Fine, so it's a double check here.

Reply:
This is the real check, and the check in ddl.api  may get false positive, we can't ensure the name don't be occupied before entering the ddl queue.